mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -1399,34 +1395,33 @@ impl MakerSwap {
             let search_input = SearchForSwapTxSpendInput {
                 time_lock: timelock,
                 other_pub: other_taker_coin_htlc_pub.as_slice(),
-                secret_hash,
+                secret_hash: &selfi.secret_hash(),
                 tx: taker_payment_hex,
                 search_from_block: taker_coin_start_block,
                 swap_contract_address: &taker_coin_swap_contract_address,
                 swap_unique_data: &unique_data,
                 watcher_reward,
             };
-            // check if the taker payment is not spent yet
+            // Check the taker payment status first
             match selfi.taker_coin.search_for_swap_tx_spend_other(search_input).await {
                 Ok(Some(FoundSwapTxSpend::Spent(tx))) => {
-                    return ERR!(
-                        "Taker payment was already spent by {} tx {:02x}",
-                        selfi.taker_coin.ticker(),
-                        tx.tx_hash_as_bytes()
-                    )
+                    // We already spent the taker payment.
+                    return Ok(tx);
                 },
                 Ok(Some(FoundSwapTxSpend::Refunded(tx))) => {
-                    return ERR!(
+                    // The taker refunded their payment.
+                    warn!("TakerPayment was refunded back to the taker.");
+                    return Err(RecoverSwapError::Irrecoverable(format!(
                         "Taker payment was already refunded by {} tx {:02x}",
                         selfi.taker_coin.ticker(),
                         tx.tx_hash_as_bytes()
-                    )
+                    )));
                 },
-                Err(e) => return ERR!("Error {} when trying to find taker payment spend", e),
+                Err(e) => return Err(RecoverSwapError::Temporary(e)),
```

Comment:
but the returned error is a string. how would tracing work here? or you mean change the error from `search_for_swap_tx_spend_other`?

Reply:
not just String, location is also prepended 
https://docs.rs/gstuff/latest/gstuff/macro.ERR.html

Reply:
> Yeah, but why not MmError? I mean instead of reverting?

Legacy code doesnt support MmError, it requires String in a lot of places

Well @mariocynicys could use MmError for internal function and for general function which returns String convert mmerr to string, but its better to just come back to ERR macro and wrap errors in internal function

Reply:
there's already an old issue to replace macros from gstuff with `MmError` generally.. 
https://github.com/KomodoPlatform/komodo-defi-framework/issues/1250

You can simply convert to string in the function caller if it requires returning `String` as Error

Reply:
> Yeah, but why not MmError? I mean instead of reverting?

ah sure, i forgot the original comment was for MmError xD

Reply:
[df94b09](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2280/commits/df94b09efc4870935a8ee7990325a7abc0f161e7)